### PR TITLE
Speed up taumol swrad

### PR DESCRIPTION
**Pull request by YakelynRJ**
This PR speeds up taumol using @jit. It validates and taumol computing time decreased from 5  to 3 seconds

1. Dataarray inputs were converted to numpy arrays 
2. taumol* were moved up to be functions 
3. taumol* were jitted 

Coverage reports (updated automatically):
- test_unit: [83%](https:\/\/output.circle-artifacts.com\/output\/job\/72afbc07-b461-4e0f-b9a4-dc707e600b2c\/artifacts\/0\/tmp\/coverage\/htmlcov-test_unit\/index.html)

### CODE REVIEW

File: external/radiation/python/radsw/radsw_main.py [hunk 1]
```diff
@@ -3141,962 +3363,1051 @@ def taumol20(
...
-        #  --- ...  the following factor is the ratio of total o2 band intensity (lines
-        #           and mate continuum) to o2 band intensity (line only). it is needed
-        #           to adjust the optical depths since the k's include only lines.
+    def return_initdata(self):
+        outdict = {"heatfac": self.heatfac, "exp_tbl": self.exp_tbl}
+        return outdict
 
-        o2adj = 1.6
-        o2tem = 4.35e-4 / (350.0 * 2.0)
+    def swrad(
```
Comment (nbren12) [inline external/radiation/python/radsw/radsw_main.py:3523]: I expect you'll see a speedup comparable to lwrad improvement if you jit this.

File: external/radiation/python/radsw/radsw_main.py [hunk 2]
```diff
@@ -1176,1059 +1181,274 @@ def cldprop(
...
-    nspb = [1, 5, 1, 1, 1, 5, 1, 0, 1, 0, 0, 1, 5, 1]
-    # band index for sfc flux
-    idxsfc = [1, 1, 1, 1, 1, 1, 1, 1, 0, 2, 2, 2, 2, 1]
-    # band index for cld prop
-    idxebc = [5, 5, 4, 4, 3, 3, 2, 2, 1, 1, 1, 1, 1, 5]
 
-    # uv-b band index
-    nuvb = 27
+@jit(nopython=True, cache=True)
+def taumol(
```
[the remark this answers is not in the document]
Comment (YakelynRJ) [reply external/radiation/python/radsw/radsw_main.py:1186]: okay.

File: external/radiation/python/radsw/radsw_main.py (hunk 1 above)
Comment (brianhenn) [reply external/radiation/python/radsw/radsw_main.py:3523]: Next PR though?

**Review by brianhenn — COMMENTED** [review]
Thanks! For readability I agree with Noah's suggestion to refactor the taumol band functions to their own module.

File: external/radiation/python/radsw/radsw_main.py (hunk 1 above)
Comment (YakelynRJ) [reply external/radiation/python/radsw/radsw_main.py:1990]: I agree. I can try to speedup this part in another PR.

File: external/radiation/python/radsw/radsw_main.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (YakelynRJ) [reply external/radiation/python/radsw/radsw_main.py:1186]: @brianhenn I added one more commit that address this comment.

[the remark this answers is not in the document]
Comment (brianhenn) [reply external/radiation/python/radsw/radsw_main.py:1186]: I think it might be cleaner if `taumol` itself got moved also.

**Review by brianhenn — APPROVED** [review]
Looks good. See comment below.

**Pull request merged**